pylib/cqlshlib/copyutil.py

```diff
@@ -1405,8 +1405,8 @@ def __init__(self, params, target):
         super(ChildProcess, self).__init__(target=target)
         self.inpipe = params['inpipe']
         self.outpipe = params['outpipe']
-        self.inmsg = None  # must be initialized after fork on Windows
-        self.outmsg = None  # must be initialized after fork on Windows
+        self.inmsg = None
```

Comment:
See above ☝️ . No reason not to init at decl?

Reply:
I am not sure about this, I admit I know 0 about threading in Python. But by mere looking at it, I do not think this is a big deal anyway. Maybe @bowensong  may shed some light on this? But I would personaly not touch it and just moved along.

Reply:
I believe that we should move the code from `on_fork()` to `__init__()`, and may also remove the `on_fork()` method if it's no longer used.

The problem with Windows and Python `multiprocessing` is that Windows doesn't support `fork()`, therefore Python implemented a workaround. On Windows, Python `multiprocessing` library uses `pickle` to serialise everything in memory, spawn a new process, and then restores the memory content from the serialised data. The `ReceivingChannel` and `SendingChannel` objects are not serialisable because they have file descriptors (which I believe it's called a file handle on Windows) in them, therefore the code has to handle them after the fake `fork()`.

However, I'm concerned that moving the code from `on_fork()` to `__init__()` may have other unintended side effects. For example, the file descriptors (FDs) will be opened before the fork, in some edge cases the fork may never happen (e.g.: an exception raised in or just after the init, but before the fork). Where's the code responsible for closing the FDs on the parent process side? Will this cause any FD leak? This clearly requires further work to find out.

To be honest, I don't think removing the comments without addressing the above is a wise move. Future developers wouldn't have the opportunity to understand why the code is written in this way if the comment is removed.

Reply:
hi @bowensong , thanks for the very detailed answer. I can revert the removal of the comment or rather change it in such a way it would mention a ticket the work (or rather investigation into this matter) would be addressed in. I honestly think the refactoring of this would prolong / delay the merging of this branch unncessarilly.

Reply:
@smiklosovic I agree. Refactoring this shouldn't be a part of this as it will unnecessarily prolong / delay this PR. The work is ideally done in a separate PR. I personally would prefer a comment added here and make it clear why the code was written in this way, and link to a JIRA ticket for refactoring this. Or, at minimal, a comment saying the code was written in this way for Windows `fork()` compatibility and is no longer necessary.